src/crawlee/storages/_base.py

```diff
@@ -21,6 +22,16 @@ def id(self) -> str:
     def name(self) -> str | None:
         """Get the storage name."""
 
+    @property
+    @abstractmethod
+    def storage_object(self) -> StorageMetadata:
```

Comment:
Wouldn't this be more precise?

```suggestion
    def storage_object(self) -> DatasetMetadata | KeyValueStoreMetadata | RequestQueueMetadata:
```

and `_StorageMetadata` could remain private.

I am also ok with having `StorageMetadata` as a type alias for the union mentioned above. And the `_StorageMetadata` could be renamed to `_CommonStorageMetadata`, `_BaseStorageMetadata`, ...

Reply:
That's what I actually did in the beginning, but I was getting type error in _creation_managment.py

<img width="1410" alt="Screenshot 2025-03-03 at 14 52 34" src="https://github.com/user-attachments/assets/34564061-97c2-4634-87ca-ab7fdc33b100" />